### PR TITLE
Meghana reddy step 11

### PR DESCRIPTION
**Pull request by Meghana-Reddy2**
DESCRIPTION 

- Let's set up validation
- Let's set up validation
- Let's think about which validation should be added to which column.
- Let's create a migration that also sets DB constraints.
- To create only the migration file, rails generatecreate it with the command
- Let's display a validation error message on the screen.
- Let's write tests for the model for validation.

Screenshot 2024-03-01 at 14 05 21
Screenshot 2024-03-01 at 14 04 59

### CODE REVIEW

File: myapp/app/models/task.rb [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+    validates :description , length: {maximum: 255, length:{ maximum: 255, long: I18n.t('flash_msgs.long')} }
     validates :status, presence: true, inclusion: { in: STATUSES }
     validates :priority, presence: true, inclusion: { in: PRIORITY }
+    validate :duedate_validate
+    
+    def duedate_validate
+        if duedate.present? && duedate < Date.today
+          errors.add(:base ,I18n.t('flash_msgs.timeover'))
+        end
+    end
```
Comment (hayashiR1) [inline myapp/app/models/task.rb]: Please unity the space of indent to 2 🙏

File: myapp/spec/system/tasks_spec.rb [hunk 2]
```diff
@@ -79,61 +79,108 @@
...
+    task = build(:task, duedate: '2024-08-08')
+    expect(task).to be_valid
+  end
+
+  it 'invalid if task duedate is past' do
+    task = build(:task, duedate: Date.today - 1)
+    expect(task).to be_invalid
   end
+end
+=======
```
Comment (hayashiR1) [inline myapp/spec/system/tasks_spec.rb]: Raser beam? 😅

File: myapp/spec/system/tasks_spec.rb [hunk 3]
```diff
@@ -79,61 +79,108 @@
...
-        visit '/'
-        click_link('Duedate')
-        expect(page.body.index('Task1')).to be < page.body.index('Task3')
-        expect(page.body.index('Task3')).to be < page.body.index('Task2')
-        click_link('Duedate')
-        expect(page.body.index('Task2')).to be < page.body.index('Task3')
-        expect(page.body.index('Task3')).to be < page.body.index('Task1')
-      end
+  it 'valid if task duedate is future' do
+    task = build(:task, duedate: '2024-08-08')
```
Comment (hayashiR1) [inline myapp/spec/system/tasks_spec.rb]: This test will fail after August 8th, so how about like this?

```suggestion
    task = build(:task, duedate: Time.zone.tomorrow)
```

**Review by hayashiR1 — COMMENTED** [review]
I posted some comments! 🙏

File: myapp/spec/system/tasks_spec.rb (hunk 3 above)
Comment (Meghana-Reddy2) [reply myapp/spec/system/tasks_spec.rb]: Changed it. It works 😄

File: myapp/spec/system/tasks_spec.rb (hunk 2 above)
Comment (Meghana-Reddy2) [reply myapp/spec/system/tasks_spec.rb]: It was because of a merge conflict 😓 . Resolved It.

**Review by hayashiR1 — APPROVED** [review]
2292b4f6-0fed-4e74-90a4-ec597387b833

**Pull request merged**